Src/Test.UnitTests.Sarif.PatternMatcher/FingerprintTests.cs

```diff
@@ -0,0 +1,357 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+
+using FluentAssertions;
+
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher
+{
+    public class FingerprintTests
```

Comment:
>FingerprintTests [](start = 17, length = 16)

@eddynake, @jameswinkler @yongyan-gh, I'd like you to take a particular look at the patterns in these unit tests, to discuss further later.